app/services/notifications/NotificationService.scala

```diff
@@ -14,26 +14,26 @@
  * limitations under the License.
  */
 
-package services
+package services.notifications
 
 import javax.inject.{Inject, Singleton}
-import models.{Notification, NotificationDetails}
+import models.Notification
 import play.api.Logger
-import reactivemongo.bson.BSONObjectID
 import repositories.NotificationsRepository
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.xml.NodeSeq
 
 @Singleton
-class NotificationService @Inject()(repository: NotificationsRepository)(implicit ec: ExecutionContext) {
+class NotificationService @Inject()(repository: NotificationsRepository, notificationFactory: NotificationFactory)(implicit ec: ExecutionContext) {
 
   private val logger = Logger(this.getClass)
```

Comment:
Why not using trait `play.api.Logging` ? It defines a logger instance for you.

Reply:
Wasn't aware about this trait to be fair. What you can see is a pattern we use across our services. This is why it is implemented this way.